File: Nelson-Alvarado-Prado_Programacion-Avanzada-Trabajo#4/api-examples.js
/**
 * API Usage Examples - JavaScript/Node.js
 * Demonstrates how to use all REST API endpoints
 */

const fetch = require('node-fetch');

const BASE_URL = 'http://localhost:3000/api';

// ================================================================
// UTILITY FUNCTIONS
// ================================================================

/**
 * Pretty print JSON responses
 */
function printResponse(title, response) {
  console.log(`\n${'='.repeat(50)}`);
  console.log(`📋 ${title}`);
  console.log('='.repeat(50));
  console.log(JSON.stringify(response, null, 2));
}

/**
 * Make GET request
 */
async function get(endpoint) {
  try {
    const response = await fetch(`${BASE_URL}${endpoint}`);
    return await response.json();
  } catch (error) {
    console.error('GET Error:', error);
    return null;
  }
}

/**
 * Make POST request
 */
async function post(endpoint, data) {
  try {
    const response = await fetch(`${BASE_URL}${endpoint}`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(data)
    });
    return await response.json();
  } catch (error) {
    console.error('POST Error:', error);
    return null;
  }
}

/**
 * Make PUT request
 */
async function put(endpoint, data) {
  try {
    const response = await fetch(`${BASE_URL}${endpoint}`, {
      method: 'PUT',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(data)
    });
    return await response.json();
  } catch (error) {
    console.error('PUT Error:', error);
    return null;
  }
}

/**
 * Make DELETE request
 */
async function del(endpoint) {
  try {
    const response = await fetch(`${BASE_URL}${endpoint}`, {
      method: 'DELETE'
    });
    return await response.json();
  } catch (error) {
    console.error('DELETE Error:', error);
    return null;
  }
}

// ================================================================
// CAR OPERATIONS EXAMPLES
// ================================================================

/**
 * Example: Create a new car
 */
async function exampleCreateCar() {
  console.log('\n🚗 CREATING A NEW CAR');
  const newCar = {
    color: 'Silver',
    year: 2024,
    make: 'Audi A4',
    type: 'sedan'
  };
  const result = await post('/cars', newCar);
  printResponse('Create Car Response', result);
  return result.data?.id;
}

/**
 * Example: Get all cars
 */
async function exampleGetAllCars() {
  console.log('\n🚗 GETTING ALL CARS');
  const result = await get('/cars');
  printResponse('Get All Cars Response', result);
  return result.data;
}

/**
 * Example: Get car by ID
 */
async function exampleGetCarById(carId) {
  console.log(`\n🚗 GETTING CAR BY ID: ${carId}`);
  const result = await get(`/cars/${carId}`);
  printResponse('Get Car by ID Response', result);
}

/**
 * Example: Get cars by color
 */
async function exampleGetCarsByColor(color) {
  console.log(`\n🚗 GETTING CARS BY COLOR: ${color}`);
  const result = await get(`/cars/filter/color/${color}`);
  printResponse('Get Cars by Color Response', result);
}

/**
 * Example: Get cars by make
 */
async function exampleGetCarsByMake(make) {
  console.log(`\n🚗 GETTING CARS BY MAKE: ${make}`);
  const result = await get(`/cars/filter/make/${encodeURIComponent(make)}`);
  printResponse('Get Cars by Make Response', result);
}

/**
 * Example: Get cars by type
 */
async function exampleGetCarsByType(type) {
  console.log(`\n🚗 GETTING CARS BY TYPE: ${type}`);
  const result = await get(`/cars/filter/type/${type}`);
  printResponse('Get Cars by Type Response', result);
}

/**
 * Example: Get cars by year range
 */
async function exampleGetCarsByYearRange(minYear, maxYear) {
  console.log(`\n🚗 GETTING CARS BY YEAR RANGE: ${minYear}-${maxYear}`);
  const result = await get(`/cars/filter/year-range?min=${minYear}&max=${maxYear}`);
  printResponse('Get Cars by Year Range Response', result);
}

/**
 * Example: Get cars with advanced filter
 */
async function exampleGetCarsAdvancedFilter() {
  console.log('\n🚗 GETTING CARS WITH ADVANCED FILTER');
  const params = new URLSearchParams({
    color: 'Red',
    minYear: 2020,
    maxYear: 2025,
    type: 'sedan'
  });
  const result = await get(`/cars/filter/advanced?${params}`);
  printResponse('Get Cars Advanced Filter Response', result);
}

/**
 * Example: Update a car
 */
async function exampleUpdateCar(carId) {
  console.log(`\n🚗 UPDATING CAR: ${carId}`);
  const updateData = {
    color: 'Gold',
    year: 2024,
    make: 'Audi A6',
    type: 'sedan'
  };
  const result = await put(`/cars/${carId}`, updateData);
  printResponse('Update Car Response', result);
}

/**
 * Example: Delete a car
 */
async function exampleDeleteCar(carId) {
  console.log(`\n🚗 DELETING CAR: ${carId}`);
  const result = await del(`/cars/${carId}`);
  printResponse('Delete Car Response', result);
}

// ================================================================
// PARKING OPERATIONS EXAMPLES
// ================================================================

/**
 * Example: Create a new parking facility
 */
async function exampleCreateParking() {
  console.log('\n🅿️  CREATING A NEW PARKING FACILITY');
  const newParking = {
    province_name: 'Murcia',
    name: 'Parking Aeropuerto',
    hourly_rate: 3.75
  };
  const result = await post('/parking', newParking);
  printResponse('Create Parking Response', result);
  return result.data?.id;
}

/**
 * Example: Get all parking facilities
 */
async function exampleGetAllParking() {
  console.log('\n🅿️  GETTING ALL PARKING FACILITIES');
  const result = await get('/parking');
  printResponse('Get All Parking Response', result);
  return result.data;
}

/**
 * Example: Get parking by ID
 */
async function exampleGetParkingById(parkingId) {
  console.log(`\n🅿️  GETTING PARKING BY ID: ${parkingId}`);
  const result = await get(`/parking/${parkingId}`);
  printResponse('Get Parking by ID Response', result);
}

/**
 * Example: Get parking by province
 */
async function exampleGetParkingByProvince(province) {
  console.log(`\n🅿️  GETTING PARKING BY PROVINCE: ${province}`);
  const result = await get(`/parking/filter/province/${province}`);
  printResponse('Get Parking by Province Response', result);
}

/**
 * Example: Get parking by name
 */
async function exampleGetParkingByName(name) {
  console.log(`\n🅿️  GETTING PARKING BY NAME: ${name}`);
  const result = await get(`/parking/filter/name/${encodeURIComponent(name)}`);
  printResponse('Get Parking by Name Response', result);
}

/**
 * Example: Get parking by rate range
 */
async function exampleGetParkingByRateRange(minRate, maxRate) {
  console.log(`\n🅿️  GETTING PARKING BY RATE RANGE: €${minRate}-€${maxRate}`);
  const result = await get(`/parking/filter/rate-range?min=${minRate}&max=${maxRate}`);
  printResponse('Get Parking by Rate Range Response', result);
}

/**
 * Example: Get parking with advanced filter
 */
async function exampleGetParkingAdvancedFilter() {
  console.log('\n🅿️  GETTING PARKING WITH ADVANCED FILTER');
  const params = new URLSearchParams({
    province: 'Madrid',
    minRate: 3.0,
    maxRate: 4.5
  });
  const result = await get(`/parking/filter/advanced?${params}`);
  printResponse('Get Parking Advanced Filter Response', result);
}

/**
 * Example: Update a parking facility
 */
async function exampleUpdateParking(parkingId) {
  console.log(`\n🅿️  UPDATING PARKING: ${parkingId}`);
  const updateData = {
    province_name: 'Murcia',
    name: 'Parking Aeropuerto Premium',
    hourly_rate: 4.50
  };
  const result = await put(`/parking/${parkingId}`, updateData);
  printResponse('Update Parking Response', result);
}

/**
 * Example: Delete a parking facility
 */
async function exampleDeleteParking(parkingId) {
  console.log(`\n🅿️  DELETING PARKING: ${parkingId}`);
  const result = await del(`/parking/${parkingId}`);
  printResponse('Delete Parking Response', result);
}

// ================================================================
// CAR ENTRY OPERATIONS EXAMPLES
// ================================================================

/**
 * Example: Create a new car entry
 */
async function exampleCreateCarEntry() {
  console.log('\n🚙 CREATING A NEW CAR ENTRY');
  const newEntry = {
    parking_id: 1,
    car_id: 1,
    entry_date_time: '2026-04-17T10:00:00',
    exit_date_time: null
  };
  const result = await post('/car-entry', newEntry);
  printResponse('Create Car Entry Response', result);
  return result.data?.sequential_number;
}

/**
 * Example: Get all car entries
 */
async function exampleGetAllCarEntries() {
  console.log('\n🚙 GETTING ALL CAR ENTRIES');
  const result = await get('/car-entry');
  printResponse('Get All Car Entries Response', result);
  return result.data;
}

/**
 * Example: Get car entry by ID
 */
async function exampleGetCarEntryById(entryId) {
  console.log(`\n🚙 GETTING CAR ENTRY BY ID: ${entryId}`);
  const result = await get(`/car-entry/${entryId}`);
  printResponse('Get Car Entry by ID Response', result);
}

/**
 * Example: Get car entries by parking
 */
async function exampleGetEntriesByParking(parkingId) {
  console.log(`\n🚙 GETTING ENTRIES FOR PARKING: ${parkingId}`);
  const result = await get(`/car-entry/filter/parking/${parkingId}`);
  printResponse('Get Entries by Parking Response', result);
}

/**
 * Example: Get car entries by car type in date range
 */
async function exampleGetEntriesByCarType(carType) {
  console.log(`\n🚙 GETTING ENTRIES FOR CAR TYPE: ${carType}`);
  const params = new URLSearchParams({
    start: '2026-04-10',
    end: '2026-04-20'
  });
  const result = await get(`/car-entry/filter/car-type/${carType}?${params}`);
  printResponse('Get Entries by Car Type Response', result);
}

/**
 * Example: Get car entries by province in date range
 */
async function exampleGetEntriesByProvince(province) {
  console.log(`\n🚙 GETTING ENTRIES FOR PROVINCE: ${province}`);
  const params = new URLSearchParams({
    start: '2026-04-10',
    end: '2026-04-20'
  });
  const result = await get(`/car-entry/filter/province/${province}?${params}`);
  printResponse('Get Entries by Province Response', result);
}

/**
 * Example: Get hourly price for parking
 */
async function exampleGetHourlyPrice(parkingId) {
  console.log(`\n🚙 GETTING HOURLY PRICE FOR PARKING: ${parkingId}`);
  const result = await get(`/car-entry/price/${parkingId}`);
  printResponse('Get Hourly Price Response', result);
}

/**
 * Example: Update a car entry
 */
async function exampleUpdateCarEntry(entryId) {
  console.log(`\n🚙 UPDATING CAR ENTRY: ${entryId}`);
  const updateData = {
    parking_id: 1,
    car_id: 1,
    entry_date_time: '2026-04-17T10:00:00',
    exit_date_time: '2026-04-17T13:00:00'
  };
  const result = await put(`/car-entry/${entryId}`, updateData);
  printResponse('Update Car Entry Response', result);
}

/**
 * Example: Delete a car entry
 */
async function exampleDeleteCarEntry(entryId) {
  console.log(`\n🚙 DELETING CAR ENTRY: ${entryId}`);
  const result = await del(`/car-entry/${entryId}`);
  printResponse('Delete Car Entry Response', result);
}

// ================================================================
// SYSTEM ENDPOINTS EXAMPLES
// ================================================================

/**
 * Example: Health check
 */
async function exampleHealthCheck() {
  console.log('\n💚 HEALTH CHECK');
  const result = await get('/health');
  printResponse('Health Check Response', result);
}

/**
 * Example: Get all endpoints
 */
async function exampleGetEndpoints() {
  console.log('\n📋 GET ALL ENDPOINTS');
  const result = await get('/endpoints');
  printResponse('Endpoints List', result);
}

// ================================================================
// MAIN EXECUTION
// ================================================================

/**
 * Run all examples
 */
async function runAllExamples() {
  try {
    console.log(`
╔════════════════════════════════════════╗
║   Car Park REST API - JavaScript       ║
║           Usage Examples               ║
╚════════════════════════════════════════╝
    `);

    // System endpoints
    await exampleHealthCheck();
    await exampleGetEndpoints();

    // Car operations
    const newCarId = await exampleCreateCar();
    await exampleGetAllCars();
    if (newCarId) await exampleGetCarById(newCarId);
    await exampleGetCarsByColor('Red');
    await exampleGetCarsByMake('Toyota Corolla');
    await exampleGetCarsByType('sedan');
    await exampleGetCarsByYearRange(2020, 2025);
    await exampleGetCarsAdvancedFilter();
    if (newCarId) await exampleUpdateCar(newCarId);

    // Parking operations
    const newParkingId = await exampleCreateParking();
    await exampleGetAllParking();
    if (newParkingId) await exampleGetParkingById(newParkingId);
    await exampleGetParkingByProvince('Madrid');
    await exampleGetParkingByRateRange(2.0, 4.0);
    await exampleGetParkingAdvancedFilter();
    if (newParkingId) await exampleUpdateParking(newParkingId);

    // Car entry operations
    const newEntryId = await exampleCreateCarEntry();
    await exampleGetAllCarEntries();
    if (newEntryId) await exampleGetCarEntryById(newEntryId);
    await exampleGetEntriesByParking(1);
    await exampleGetEntriesByCarType('sedan');
    await exampleGetEntriesByProvince('Madrid');
    await exampleGetHourlyPrice(1);
    if (newEntryId) await exampleUpdateCarEntry(newEntryId);

    console.log(`
╔════════════════════════════════════════╗
║        Examples Completed! ✅           ║
╚════════════════════════════════════════╝
    `);
  } catch (error) {
    console.error('❌ Error running examples:', error);
  }
}

// Run examples if this file is executed directly
if (require.main === module) {
  runAllExamples();
}

module.exports = {
  get, post, put, del,
  exampleCreateCar, exampleGetAllCars, exampleGetCarById,
  exampleCreateParking, exampleGetAllParking,
  exampleCreateCarEntry, exampleGetAllCarEntries,
  runAllExamples
};

File: Nelson-Alvarado-Prado_Programacion-Avanzada-Trabajo#4/api_examples.py
"""
API Usage Examples - Python
Demonstrates how to use all REST API endpoints
"""

import requests
import json
from datetime import datetime, timedelta

BASE_URL = 'http://localhost:5000/api'

# ================================================================
# UTILITY FUNCTIONS
# ================================================================

def print_response(title, response):
    """Pretty print JSON responses"""
    print(f"\n{'='*60}")
    print(f"📋 {title}")
    print('='*60)
    print(json.dumps(response, indent=2, ensure_ascii=False))

def make_request(method, endpoint, data=None):
    """Make HTTP request and return JSON"""
    try:
        url = f"{BASE_URL}{endpoint}"
        if method == 'GET':
            response = requests.get(url)
        elif method == 'POST':
            response = requests.post(url, json=data)
        elif method == 'PUT':
            response = requests.put(url, json=data)
        elif method == 'DELETE':
            response = requests.delete(url)
        
        if response.status_code in [200, 201]:
            return response.json()
        else:
            print(f"❌ Error {response.status_code}: {response.text}")
            return None
    except Exception as e:
        print(f"❌ Request Error: {e}")
        return None

def get(endpoint):
    """Make GET request"""
    return make_request('GET', endpoint)

def post(endpoint, data):
    """Make POST request"""
    return make_request('POST', endpoint, data)

def put(endpoint, data):
    """Make PUT request"""
    return make_request('PUT', endpoint, data)

def delete(endpoint):
    """Make DELETE request"""
    return make_request('DELETE', endpoint)

# ================================================================
# CAR OPERATIONS EXAMPLES
# ================================================================

def example_create_car():
    """Example: Create a new car"""
    print("\n🚗 CREATING A NEW CAR")
    new_car = {
        'color': 'Silver',
        'year': 2024,
        'make': 'Audi A4',
        'type': 'sedan'
    }
    result = post('/cars', new_car)
    print_response('Create Car Response', result)
    return result['data']['id'] if result and result.get('success') else None

def example_get_all_cars():
    """Example: Get all cars"""
    print("\n🚗 GETTING ALL CARS")
    result = get('/cars')
    print_response('Get All Cars Response', result)
    return result['data'] if result else None

def example_get_car_by_id(car_id):
    """Example: Get car by ID"""
    print(f"\n🚗 GETTING CAR BY ID: {car_id}")
    result = get(f'/cars/{car_id}')
    print_response('Get Car by ID Response', result)

def example_get_cars_by_color(color):
    """Example: Get cars by color"""
    print(f"\n🚗 GETTING CARS BY COLOR: {color}")
    result = get(f'/cars/filter/color/{color}')
    print_response('Get Cars by Color Response', result)

def example_get_cars_by_make(make):
    """Example: Get cars by make"""
    print(f"\n🚗 GETTING CARS BY MAKE: {make}")
    result = get(f'/cars/filter/make/{make.replace(" ", "%20")}')
    print_response('Get Cars by Make Response', result)

def example_get_cars_by_type(car_type):
    """Example: Get cars by type"""
    print(f"\n🚗 GETTING CARS BY TYPE: {car_type}")
    result = get(f'/cars/filter/type/{car_type}')
    print_response('Get Cars by Type Response', result)

def example_get_cars_by_year_range(min_year, max_year):
    """Example: Get cars by year range"""
    print(f"\n🚗 GETTING CARS BY YEAR RANGE: {min_year}-{max_year}")
    result = get(f'/cars/filter/year-range?min={min_year}&max={max_year}')
    print_response('Get Cars by Year Range Response', result)

def example_get_cars_advanced_filter():
    """Example: Get cars with advanced filter"""
    print("\n🚗 GETTING CARS WITH ADVANCED FILTER")
    params = {
        'color': 'Red',
        'minYear': 2020,
        'maxYear': 2025,
        'type': 'sedan'
    }
    query_string = '&'.join([f"{k}={v}" for k, v in params.items()])
    result = get(f'/cars/filter/advanced?{query_string}')
    print_response('Get Cars Advanced Filter Response', result)

def example_update_car(car_id):
    """Example: Update a car"""
    print(f"\n🚗 UPDATING CAR: {car_id}")
    update_data = {
        'color': 'Gold',
        'year': 2024,
        'make': 'Audi A6',
        'type': 'sedan'
    }
    result = put(f'/cars/{car_id}', update_data)
    print_response('Update Car Response', result)

def example_delete_car(car_id):
    """Example: Delete a car"""
    print(f"\n🚗 DELETING CAR: {car_id}")
    result = delete(f'/cars/{car_id}')
    print_response('Delete Car Response', result)

# ================================================================
# PARKING OPERATIONS EXAMPLES
# ================================================================

def example_create_parking():
    """Example: Create a new parking facility"""
    print("\n🅿️  CREATING A NEW PARKING FACILITY")
    new_parking = {
        'province_name': 'Murcia',
        'name': 'Parking Aeropuerto',
        'hourly_rate': 3.75
    }
    result = post('/parking', new_parking)
    print_response('Create Parking Response', result)
    return result['data']['id'] if result and result.get('success') else None

def example_get_all_parking():
    """Example: Get all parking facilities"""
    print("\n🅿️  GETTING ALL PARKING FACILITIES")
    result = get('/parking')
    print_response('Get All Parking Response', result)
    return result['data'] if result else None

def example_get_parking_by_id(parking_id):
    """Example: Get parking by ID"""
    print(f"\n🅿️  GETTING PARKING BY ID: {parking_id}")
    result = get(f'/parking/{parking_id}')
    print_response('Get Parking by ID Response', result)

def example_get_parking_by_province(province):
    """Example: Get parking by province"""
    print(f"\n🅿️  GETTING PARKING BY PROVINCE: {province}")
    result = get(f'/parking/filter/province/{province}')
    print_response('Get Parking by Province Response', result)

def example_get_parking_by_name(name):
    """Example: Get parking by name"""
    print(f"\n🅿️  GETTING PARKING BY NAME: {name}")
    result = get(f'/parking/filter/name/{name.replace(" ", "%20")}')
    print_response('Get Parking by Name Response', result)

def example_get_parking_by_rate_range(min_rate, max_rate):
    """Example: Get parking by rate range"""
    print(f"\n🅿️  GETTING PARKING BY RATE RANGE: €{min_rate}-€{max_rate}")
    result = get(f'/parking/filter/rate-range?min={min_rate}&max={max_rate}')
    print_response('Get Parking by Rate Range Response', result)

def example_get_parking_advanced_filter():
    """Example: Get parking with advanced filter"""
    print("\n🅿️  GETTING PARKING WITH ADVANCED FILTER")
    params = {
        'province': 'Madrid',
        'minRate': 3.0,
        'maxRate': 4.5
    }
    query_string = '&'.join([f"{k}={v}" for k, v in params.items()])
    result = get(f'/parking/filter/advanced?{query_string}')
    print_response('Get Parking Advanced Filter Response', result)

def example_update_parking(parking_id):
    """Example: Update a parking facility"""
    print(f"\n🅿️  UPDATING PARKING: {parking_id}")
    update_data = {
        'province_name': 'Murcia',
        'name': 'Parking Aeropuerto Premium',
        'hourly_rate': 4.50
    }
    result = put(f'/parking/{parking_id}', update_data)
    print_response('Update Parking Response', result)

def example_delete_parking(parking_id):
    """Example: Delete a parking facility"""
    print(f"\n🅿️  DELETING PARKING: {parking_id}")
    result = delete(f'/parking/{parking_id}')
    print_response('Delete Parking Response', result)

# ================================================================
# CAR ENTRY OPERATIONS EXAMPLES
# ================================================================

def example_create_car_entry():
    """Example: Create a new car entry"""
    print("\n🚙 CREATING A NEW CAR ENTRY")
    new_entry = {
        'parking_id': 1,
        'car_id': 1,
        'entry_date_time': '2026-04-17T10:00:00',
        'exit_date_time': None
    }
    result = post('/car-entry', new_entry)
    print_response('Create Car Entry Response', result)
    return result['data']['sequential_number'] if result and result.get('success') else None

def example_get_all_car_entries():
    """Example: Get all car entries"""
    print("\n🚙 GETTING ALL CAR ENTRIES")
    result = get('/car-entry')
    print_response('Get All Car Entries Response', result)
    return result['data'] if result else None

def example_get_car_entry_by_id(entry_id):
    """Example: Get car entry by ID"""
    print(f"\n🚙 GETTING CAR ENTRY BY ID: {entry_id}")
    result = get(f'/car-entry/{entry_id}')
    print_response('Get Car Entry by ID Response', result)

def example_get_entries_by_parking(parking_id):
    """Example: Get car entries by parking"""
    print(f"\n🚙 GETTING ENTRIES FOR PARKING: {parking_id}")
    result = get(f'/car-entry/filter/parking/{parking_id}')
    print_response('Get Entries by Parking Response', result)

def example_get_entries_by_car_type(car_type):
    """Example: Get car entries by car type in date range"""
    print(f"\n🚙 GETTING ENTRIES FOR CAR TYPE: {car_type}")
    params = {
        'start': '2026-04-10',
        'end': '2026-04-20'
    }
    query_string = '&'.join([f"{k}={v}" for k, v in params.items()])
    result = get(f'/car-entry/filter/car-type/{car_type}?{query_string}')
    print_response('Get Entries by Car Type Response', result)

def example_get_entries_by_province(province):
    """Example: Get car entries by province in date range"""
    print(f"\n🚙 GETTING ENTRIES FOR PROVINCE: {province}")
    params = {
        'start': '2026-04-10',
        'end': '2026-04-20'
    }
    query_string = '&'.join([f"{k}={v}" for k, v in params.items()])
    result = get(f'/car-entry/filter/province/{province}?{query_string}')
    print_response('Get Entries by Province Response', result)

def example_get_hourly_price(parking_id):
    """Example: Get hourly price for parking"""
    print(f"\n🚙 GETTING HOURLY PRICE FOR PARKING: {parking_id}")
    result = get(f'/car-entry/price/{parking_id}')
    print_response('Get Hourly Price Response', result)

def example_update_car_entry(entry_id):
    """Example: Update a car entry"""
    print(f"\n🚙 UPDATING CAR ENTRY: {entry_id}")
    update_data = {
        'parking_id': 1,
        'car_id': 1,
        'entry_date_time': '2026-04-17T10:00:00',
        'exit_date_time': '2026-04-17T13:00:00'
    }
    result = put(f'/car-entry/{entry_id}', update_data)
    print_response('Update Car Entry Response', result)

def example_delete_car_entry(entry_id):
    """Example: Delete a car entry"""
    print(f"\n🚙 DELETING CAR ENTRY: {entry_id}")
    result = delete(f'/car-entry/{entry_id}')
    print_response('Delete Car Entry Response', result)

# ================================================================
# SYSTEM ENDPOINTS EXAMPLES
# ================================================================

def example_health_check():
    """Example: Health check"""
    print("\n💚 HEALTH CHECK")
    result = get('/health')
    print_response('Health Check Response', result)

def example_get_endpoints():
    """Example: Get all endpoints"""
    print("\n📋 GET ALL ENDPOINTS")
    result = get('/endpoints')
    print_response('Endpoints List', result)

# ================================================================
# MAIN EXECUTION
# ================================================================

def run_all_examples():
    """Run all examples"""
    try:
        print("""
╔════════════════════════════════════════╗
║   Car Park REST API - Python           ║
║           Usage Examples               ║
╚════════════════════════════════════════╝
        """)

        # System endpoints
        example_health_check()
        example_get_endpoints()

        # Car operations
        new_car_id = example_create_car()
        example_get_all_cars()
        if new_car_id:
            example_get_car_by_id(new_car_id)
        example_get_cars_by_color('Red')
        example_get_cars_by_make('Toyota Corolla')
        example_get_cars_by_type('sedan')
        example_get_cars_by_year_range(2020, 2025)
        example_get_cars_advanced_filter()
        if new_car_id:
            example_update_car(new_car_id)

        # Parking operations
        new_parking_id = example_create_parking()
        example_get_all_parking()
        if new_parking_id:
            example_get_parking_by_id(new_parking_id)
        example_get_parking_by_province('Madrid')
        example_get_parking_by_rate_range(2.0, 4.0)
        example_get_parking_advanced_filter()
        if new_parking_id:
            example_update_parking(new_parking_id)

        # Car entry operations
        new_entry_id = example_create_car_entry()
        example_get_all_car_entries()
        if new_entry_id:
            example_get_car_entry_by_id(new_entry_id)
        example_get_entries_by_parking(1)
        example_get_entries_by_car_type('sedan')
        example_get_entries_by_province('Madrid')
        example_get_hourly_price(1)
        if new_entry_id:
            example_update_car_entry(new_entry_id)

        print("""
╔════════════════════════════════════════╗
║        Examples Completed! ✅           ║
╚════════════════════════════════════════╝
        """)

    except Exception as error:
        print(f"❌ Error running examples: {error}")

if __name__ == '__main__':
    run_all_examples()
